safe_control_gym/controllers/lqr/ilqr.py

```diff
@@ -299,6 +309,9 @@ def select_action(self, obs, info=None, training=False):
         else:
             action, _, _ = self.calculate_lqr_action(obs, step)
 
+        if hardware and self.traj_step < self.num_steps - 1:
```

Comment:
I have reverted back the changes to ilqr and correspondingly fixed the config file to avoid further issue. We will fix it in the next pr.

Reply:
Hi Fed. I will take care of the ilqr part in this PR. `self.num_steps` is computed after the input stack is obtained and I think it will depends on how much steps are run in the `run` method. I suppose `run` method can terminate early such that `sefl.num_steps` is less than `max_steps`.